### PR TITLE
🌐 Add Turkish translation for `docs/tr/docs/history-design-future.md`

### PR DESCRIPTION
**Pull request by hasansezertasan**
🌐 Add Turkish translation for `docs/tr/docs/history-design-future.md`

[Original File](https://github.com/tiangolo/fastapi/blob/master/docs/en/docs/history-design-future.md)

Discussion: #9193

### CODE REVIEW

**Comment by hasansezertasan** [comment]
@alperiox, @OzgunCaglarArslan, can you please leave a review?

**Comment by tiangolo** [comment]
📝 Docs preview for commit d38dd625c502e7bab988943bbb3b825d0ec1c111 at: https://89e38dd4.fastapitiangolo.pages.dev

**Comment by hasansezertasan** [comment]
@alperiox, @OzgunCaglarArslan @SametEmin, @mertssmnoglu, can you please leave a review?

File: docs/tr/docs/history-design-future.md [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+
+Sonuç olarak, **FastAPI**'ı kodlamaya başlamadan önce birkaç ay boyunca, OpenAPI, JSON Schema, OAuth2 ve benzerlerinin tanımlamalarını inceledim. İlişkilerini, örtüşdükleri noktaları ve farklılıklarını anlamaya çalıştım.
+
+## Tasarım
+
+Sonrasında, (**FastAPI** kullanan bir geliştirici olarak) sahip olmak istediğim "API"ı tasarlamak için biraz zaman harcadım.
+
+Çeşitli fikirleri en popüler Python editörlerinde test ettim: PyCharm, VS Code, Jedi tabanlı editörler.
+
+En son <a href="https://www.jetbrains.com/research/python-developers-survey-2018/#development-tools" class="external-link" target="_blank">Python Developer Survey</a>'ine göre, kullanıcıların yaklaşık %80'inin kullandığı editörler.
```
[the remark this answers is not in the document]
Comment (mertssmnoglu) [reply docs/tr/docs/history-design-future.md:47]: "Bu editörler, en son yapılan Python Developer Survey'ine göre kullanıcıların yaklaşık %80'inin kullandığı editörler."
With this sentence, we can connect previous and after sentences.

File: docs/tr/docs/history-design-future.md [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+
+Bu demek oluyor ki **FastAPI**, Python geliştiricilerinin %80'inin kullandığı editörlerle test edildi. Ve diğer editörlerin çoğu benzer şekilde çalıştığından, bütün artıları neredeyse tüm editörlerde çalışacaktır.
```
Comment (mertssmnoglu) [inline docs/tr/docs/history-design-future.md:49]: According to the change that I prefer on the previous comment, the beginning can change to "Bu da demek oluyor ki..."

**Review by mertssmnoglu — CHANGES_REQUESTED** [review]
Clear explanation. 
As general it's very good. But it can be more understandable the changes that i prefer. What do you think about them? @hasansezertasan

File: docs/tr/docs/history-design-future.md (hunk 2 above)
Comment (hasansezertasan) [reply docs/tr/docs/history-design-future.md:49]: Thank you for the review. I recommend you make a "suggestion" block for these "obvious" mistakes.

Take a look at this link: [How can I manually add suggestions in code reviews on GitHub? - Stack Overflow](https://stackoverflow.com/questions/54239733/)

Thank you for this catch, I'll update it!

File: docs/tr/docs/history-design-future.md [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+
+Yıllardır yeni bir framework oluşturmaktan kaçınıyordum. Başlangıçta **FastAPI**'ın çözdüğü sorunları çözebilmek için pek çok farklı framework, <abbr title="Eklenti: Plug-In">eklenti</abbr> ve araç kullanmayı denedim.
+
+Ancak bir noktada, geçmişteki diğer araçlardan en iyi fikirleri alarak bütün bu çözümleri kapsayan, ayrıca bütün bunları Python'ın daha önce mevcut olmayan özelliklerini (Python 3.6+ ile gelen <abbr title="Tip belirteçleri: Type Hints">tip belirteçleri</abbr>) kullanarak yapan bir şey üretmekten başka bir seçenek kalmamıştı.
+
+</blockquote>
+
+## Araştırma
+
+Önceki alternatifleri kullanarak, hepsinden öğrenme şansım oldu, fikirleri alıp, kendim ve çalıştığım geliştirici ekipleri için en iyi şekilde birleştirebildim.
```
Comment (alperiox) [inline docs/tr/docs/history-design-future.md:33]:
```suggestion
Önceki alternatifleri kullanarak hepsinden bir şeyler öğrenip, fikirler alıp, bunları kendim ve çalıştığım geliştirici ekipler için en iyi şekilde birleştirebilme şansım oldu.
```

File: docs/tr/docs/history-design-future.md [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+
+Mesela, ideal olarak standart Python tip belirteçlerine dayanması gerektiği açıktı.
+
+Ayrıca, en iyi yaklaşım zaten mevcut olan standartları kullanmaktı.
+
+Sonuç olarak, **FastAPI**'ı kodlamaya başlamadan önce birkaç ay boyunca, OpenAPI, JSON Schema, OAuth2 ve benzerlerinin tanımlamalarını inceledim. İlişkilerini, örtüşdükleri noktaları ve farklılıklarını anlamaya çalıştım.
```
Comment (alperiox) [inline docs/tr/docs/history-design-future.md:39]:
```suggestion
Sonuç olarak, **FastAPI**'ı kodlamaya başlamadan önce birkaç ay boyunca, OpenAPI, JSON Schema, OAuth2 ve benzerlerinin tanımlamalarını inceledim. İlişkilerini, örtüştükleri noktaları ve farklılıklarını anlamaya çalıştım.
```

File: docs/tr/docs/history-design-future.md [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+
+Çeşitli alternatifleri test ettikten sonra, avantajlarından dolayı <a href="https://pydantic-docs.helpmanual.io/" class="external-link" target="_blank">**Pydantic**</a>'i kullanmaya karar verdim.
+
+Sonra, JSON Schema ile tamamen uyumlu olmasını sağlamak, kısıtlama bildirimlerini tanımlamanın farklı yollarını desteklemek ve birkaç editördeki testlere dayanarak editör desteğini (tip kontrolleri, otomatik tamamlama) geliştirmek için katkıda bulundum.
+
+Geliştirme sırasında, diğer ana gereksinim olan <a href="https://www.starlette.io/" class="external-link" target="_blank">**Starlette**</a>'e de katkıda bulundum.
+
+## Geliştirme
+
+**FastAPI**'ı oluşturmaya başladığımda, parçaların çoğu zaten yerindeydi, tasarım tanımlanmıştı, gereksinimler ve araçlar hazırdı ve standartlar ve tanımlar hakkındaki bilgi net ve tazeydi.
```
Comment (alperiox) [inline docs/tr/docs/history-design-future.md:65]:
```suggestion
**FastAPI**'ı oluşturmaya başladığımda, parçaların çoğu zaten yerindeydi, tasarım tanımlanmıştı, gereksinimler ve araçlar hazırdı, standartlar ve tanımlamalar hakkındaki bilgi net ve tazeydi.
```

File: docs/tr/docs/history-design-future.md [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+
+## Gelecek
+
+Şimdiye kadar, **FastAPI**'ın fikirleri birçok kişiye faydalı olduğu apaçık ortada.
```
Comment (alperiox) [inline docs/tr/docs/history-design-future.md:69]:
```suggestion
Şimdiye kadar, **FastAPI**'ın fikirleriyle birçok kişiye faydalı olduğu apaçık ortada.
```

**Review by alperiox — COMMENTED** [review]
Looks pretty nice, I added minor changes that may fit better to the given context.

File: docs/tr/docs/history-design-future.md [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+
+İşte o geçmişin bir kısmı.
+
+## Alternatifler
+
+Bir süredir karmaşık gereksinimlere sahip API'lar oluşturuyor (Makine Öğrenimi, dağıtık sistemler, asenkron işler, NoSQL veritabanları vb.) ve farklı geliştirici ekiplerini yönetiyorum.
+
+Bu süreçte birçok alternatifi araştırmak, test etmek ve kullanmak zorunda kaldım.
+
+**FastAPI**'ın geçmişi, büyük ölçüde öncüllerinin geçmişini kapsıyor.
```
[the remark this answers is not in the document]
Comment (hasansezertasan) [reply docs/tr/docs/history-design-future.md:15]:
```suggestion
**FastAPI**'ın geçmişi, büyük ölçüde seleflerinin geçmişini kapsıyor.
```

Nasıl duruyor @alperiox

[the remark this answers is not in the document]
Comment (hasansezertasan) [reply docs/tr/docs/history-design-future.md:15]: Uygun görmezsen, en uygun gördüğün öneriyi gönderebilirsin, direkt ekleyeceğim.

**Review by alperiox — APPROVED** [review]
I think it is pretty accurate now, thank you for the hard work!

**Review by mertssmnoglu — APPROVED** [review]

**Comment by hasansezertasan** [comment]
> I think it is pretty accurate now, thank you for the hard work!

Thank you for the great review 🙏

**Comment by tiangolo** [comment]
📝 Docs preview for commit 6082448f82e0e7efcbfed755ef3a566f7b7ded05 at: https://491d1788.fastapitiangolo.pages.dev

File: docs/tr/docs/history-design-future.md (hunk 7 above)
[the remark this answers is not in the document]
Comment (alperiox) [reply docs/tr/docs/history-design-future.md:15]:
```suggestion
**FastAPI**'ın geçmişi, büyük ölçüde önceden geliştirilen araçların geçmişini kapsıyor.
```

Böyle bir şey daha uygun olabilir, açıkçası 'selef' bence yanlış değil ve buraya gayet uygun fakat ne kadar bilinen ve kullanılan bir kelime olduğundan emin değilim. @hasansezertasan

[the remark this answers is not in the document]
Comment (hasansezertasan) [reply docs/tr/docs/history-design-future.md:15]: Önerin uygun, commitleyeceğim. Sonrasında son bir review atabilirsen sevinirim. Çok teşekkür ederim 🎂

**Comment by tiangolo** [comment]
📝 Docs preview for commit 8415434eaad98f9b4709f4f47714345e58724a3a at: https://4f6317fd.fastapitiangolo.pages.dev

**Comment by egndz** [comment]
pretty good one and easily understandable by Turkish speakers. Cheers!

**Review by OzgunCaglarArslan — APPROVED** [review]

**Comment by hasansezertasan** [comment]
Thank you @OzgunCaglarArslan for the review!

File: docs/tr/docs/history-design-future.md [hunk 8]
```diff
@@ -0,0 +1,79 @@
...
+
+## Araştırma
+
+Önceki alternatifleri kullanarak hepsinden bir şeyler öğrenip, fikirler alıp, bunları kendim ve çalıştığım geliştirici ekipler için en iyi şekilde birleştirebilme şansım oldu.
+
+Mesela, ideal olarak standart Python tip belirteçlerine dayanması gerektiği açıktı.
+
+Ayrıca, en iyi yaklaşım zaten mevcut olan standartları kullanmaktı.
+
+Sonuç olarak, **FastAPI**'ı kodlamaya başlamadan önce birkaç ay boyunca, OpenAPI, JSON Schema, OAuth2 ve benzerlerinin tanımlamalarını inceledim. İlişkilerini, örtüştükleri noktaları ve farklılıklarını anlamaya çalıştım.
```
Comment (OzgunCaglarArslan) [inline docs/tr/docs/history-design-future.md:39]:
```suggestion
Sonuç olarak, **FastAPI**'ı kodlamaya başlamadan önce, birkaç ay boyunca OpenAPI, JSON Schema, OAuth2 vb. özelliklerini inceledim. İlişkilerini, örtüştükleri noktaları ve farklılıklarını anlamaya çalıştım.
```

File: docs/tr/docs/history-design-future.md [hunk 9]
```diff
@@ -0,0 +1,79 @@
...
+
+## Tasarım
+
+Sonrasında, (**FastAPI** kullanan bir geliştirici olarak) sahip olmak istediğim "API"ı tasarlamak için biraz zaman harcadım.
+
+Çeşitli fikirleri en popüler Python editörlerinde test ettim: PyCharm, VS Code, Jedi tabanlı editörler.
+
+Bu test, en son <a href="https://www.jetbrains.com/research/python-developers-survey-2018/#development-tools" class="external-link" target="_blank">Python Developer Survey</a>'ine göre, kullanıcıların yaklaşık %80'inin kullandığı editörleri kapsıyor.
+
+Bu da demek oluyor ki **FastAPI**, Python geliştiricilerinin %80'inin kullandığı editörlerle test edildi. Ve diğer editörlerin çoğu benzer şekilde çalıştığından, bütün artıları neredeyse tüm editörlerde çalışacaktır.
```
Comment (OzgunCaglarArslan) [inline docs/tr/docs/history-design-future.md:49]:
```suggestion
Bu da demek oluyor ki **FastAPI**, Python geliştiricilerinin %80'inin kullandığı editörlerle test edildi. Ve diğer editörlerin çoğu benzer şekilde çalıştığından, avantajları neredeyse tüm editörlerde çalışacaktır.
```

**Review by OzgunCaglarArslan — CHANGES_REQUESTED** [review]

File: docs/tr/docs/history-design-future.md (hunk 8 above)
Comment (hasansezertasan) [reply docs/tr/docs/history-design-future.md:39]: Burada "özellikleri" değilde "tanımlamaları" daha uygun bir kullanım ancak virgül düzeltmesini yapabiliriz.

```suggestion
Sonuç olarak, **FastAPI**'ı kodlamaya başlamadan önce, birkaç ay boyunca OpenAPI, JSON Schema, OAuth2 ve benzerlerinin tanımlamalarını inceledim. İlişkilerini, örtüştükleri noktaları ve farklılıklarını anlamaya çalıştım.
```

**Comment by hasansezertasan** [comment]
Thank you @OzgunCaglarArslan for the review. I fixed the parts you highlighter. Can you please take a look at it again?

**Pull request merged**